src/psyclone/tests/psyir/nodes/psy_data_node_test.py

```diff
@@ -568,59 +507,29 @@ def test_psy_data_node_name_clash(fortran_writer):
-    assert ('CALL extract_psy_data%PreDeclareVariable("f1_data@'
-            'module_with_name_clash_mod", f1_data_1)' in code)
-    assert ('CALL extract_psy_data%ProvideVariable("f1_data@'
-            'module_with_name_clash_mod", f1_data_1)' in code)
-    assert ('CALL extract_psy_data%PreDeclareVariable("f2_data@'
-            'module_with_name_clash_mod", f2_data_1)' in code)
-    assert ('CALL extract_psy_data%PreDeclareVariable("f2_data_post@'
-            'module_with_name_clash_mod", f2_data_1)' in code)
-    assert ('CALL extract_psy_data%ProvideVariable("f2_data@'
-            'module_with_name_clash_mod", f2_data_1)' in code)
-    assert ('CALL extract_psy_data%ProvideVariable("f2_data_post@'
-            'module_with_name_clash_mod", f2_data_1)' in code)
-
-    # Second test, use lower_to_language_level:
-    # -----------------------------------------
-    invoke.schedule.children[0].lower_to_language_level()
-
-    # Note that atm we cannot fortran_writer() the schedule, LFRic does not
-    # yet fully support this. So we just lower each line individually:
-    code = "".join([fortran_writer(i) for i in invoke.schedule.children])
+    code = fortran_writer(invoke.schedule)
 
     assert ('CALL extract_psy_data % PreDeclareVariable("f1_data_post", '
             'f1_data)' in code)
     assert ('CALL extract_psy_data % PreDeclareVariable("f1_data@'
             'module_with_name_clash_mod", f1_data_1)' in code)
     assert ('CALL extract_psy_data % PreDeclareVariable("f2_data@'
             'module_with_name_clash_mod", f2_data_1)' in code)
-    assert ('CALL extract_psy_data % PreDeclareVariable("f2_data@'
-            'module_with_name_clash_mod_post", f2_data_1)' in code)
+    assert ('CALL extract_psy_data % PreDeclareVariable("f2_data_post@'
+            'module_with_name_clash_mod", f2_data_1)' in code)
 
     assert ('CALL extract_psy_data % ProvideVariable("f1_data@'
             'module_with_name_clash_mod", f1_data_1)' in code)
     assert ('CALL extract_psy_data % ProvideVariable("f2_data@'
             'module_with_name_clash_mod", f2_data_1)' in code)
-    assert ('CALL extract_psy_data % ProvideVariable("f2_data@'
-            'module_with_name_clash_mod_post", f2_data_1)' in code)
+    assert ('CALL extract_psy_data % ProvideVariable("f2_data_post@'
+            'module_with_name_clash_mod", f2_data_1)' in code)
 
 
 # ----------------------------------------------------------------------------
 def test_psy_data_node_lfric_inside_of_loop():
     '''Test that if a PSyData node is inside a loop (which means the code will
-    already be generated by PSyIR), the required psydata variable is declared.
-    ATM (TODO #1010) LFRicLoop.gen_code calls a fix_gen_code function in the
-    PSyData node to add the declaration (which the PSyData node added to the
-    symbol table, but since the symbol table is not used by gen_code, it
-    would otherwise be missing).
+    already be generated by PSyIR).
```

Comment:
You've shortened this docstring a bit too much :-)

Reply:
Reverted missing part